src/transactions/transactionBytes.js

```diff
@@ -12,323 +12,252 @@
+ * @return {Buffer}
+ */
 
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
-	}
+export function getAssetDataForDappTransaction({ dapp }) {
+	checkRequiredFields(['name', 'link', 'type', 'category'], dapp);
+	const { name, description, tags, link, icon, type, category } = dapp;
+	const dappNameBuffer = Buffer.from(name, 'utf8');
+	const dappLinkBuffer = Buffer.from(link, 'utf8');
+	const dappTypeBuffer = Buffer.alloc(4, type);
+	const dappCategoryBuffer = Buffer.alloc(4, category);
+
+	const dappDescriptionBuffer = description ? Buffer.from(description, 'utf8') : Buffer.alloc(0);
+	const dappTagsBuffer = tags ? Buffer.from(tags, 'utf8') : Buffer.alloc(0);
+	const dappIconBuffer = icon ? Buffer.from(icon, 'utf8') : Buffer.alloc(0);
+
+	return Buffer.concat([
+		dappNameBuffer,
+		dappDescriptionBuffer,
+		dappTagsBuffer,
+		dappLinkBuffer,
+		dappIconBuffer,
+		dappTypeBuffer,
+		dappCategoryBuffer,
+	]);
+}
 
-	/**
-	 * @method isDappInTransferTransaction
-	 * @return {object}
-	 */
+/**
+ * @method getAssetDataForDappInTransaction
+ * @param {Object} transactionAsset
+ * @return {Buffer}
+ */
 
-	function isDappInTransferTransaction() {
-		const buf = Buffer.from(transaction.asset.inTransfer.dappId);
+export function getAssetDataForDappInTransaction({ inTransfer }) {
+	checkRequiredFields(['dappId'], inTransfer);
+	const { dappId } = inTransfer;
+	return Buffer.from(dappId, 'utf8');
+}
 
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
-	}
+/**
+ * @method getASsetDataForDappOutTransaction
+ * @param {Object} transactionAsset
+ * @return {Buffer}
+ */
 
-	/**
-	 * @method isDappOutTransferTransaction
-	 * @return {object}
-	 */
+export function getAssetDataForDappOutTransaction({ outTransfer }) {
+	checkRequiredFields(['dappId', 'transactionId'], outTransfer);
+	const { dappId, transactionId } = outTransfer;
+	const dappOutAppIdBuffer = Buffer.from(dappId, 'utf8');
+	const dappOutTransactionIdBuffer = Buffer.from(transactionId, 'utf8');
 
-	function isDappOutTransferTransaction() {
-		const dappBuf = Buffer.from(transaction.asset.outTransfer.dappId);
-		const transactionBuf = Buffer.from(transaction.asset.outTransfer.transactionId);
-		const buf = Buffer.concat([dappBuf, transactionBuf]);
+	return Buffer.concat([dappOutAppIdBuffer, dappOutTransactionIdBuffer]);
+}
 
-		return {
-			assetBytes: buf,
-			assetSize: buf.length,
-		};
-	}
+/**
+ * @method getAssetBytesHelper
+ * @param {Object} transaction
+ * @return {Buffer}
+ */
 
-	/**
-	 * `transactionType` describes the available transaction types.
-	 *
-	 * @property transactionType
-	 * @type object
-	 */
-
-	const transactionType = {
-		0: isSendTransaction,
-		1: isSignatureTransaction,
-		2: isDelegateTransaction,
-		3: isVoteTransaction,
-		4: isMultisignatureTransaction,
-		5: isDappTransaction,
-		6: isDappInTransferTransaction,
-		7: isDappOutTransferTransaction,
+export function getAssetBytesHelper(transaction) {
+	const assetDataGetters = {
```

Comment:
It looks odd to have an object whose keys are `0`–`7`, but I guess it makes more sense than an array in this case.